docs/Migration_Guide_Zephyr.md

```diff
@@ -0,0 +1,587 @@
+builds, here is a comparison of the old and new naming:
+
+```c
+/* Remove this from your existing code */
+#include <net/golioth/system_client.h>
+
+/* Replace it with this updated syntax */
+#include <golioth/client.h>
+```
+
+The naming scheme for these files has been updated so that it is easier
+to understand what includes do. This comes with the added benefit of
+making it easier to intuitively know what header file you need.
+
+### Sample header files
+
+Golioth example code uses a common sample library. This is not
+technically a part of the core SDK (it servers as a guide for developing
+your own applications based on the SDK). Header files for sample library
+largely retain the same naming scheme, though several new headers have
+been added.
+
+```c
+/* This include remains unchanged */
+#include <samples/common/net_connect.h>
+```
+
+## Change Golioth API calls
+
+Most of the API names and formats have changed. The majority of the
+migration work will be in correctly updating these calls. This section
+highlights the most commonly needed changes.
+
+> :information_source: **Compare Sample Code**: The most direct
+> comparison of these changes is found by comparing samples between the
+> deprecated [Golioth Zephyr
+> SDK](https://github.com/golioth/golioth-zephyr-sdk/tree/main/samples)
+> and the [Golioth Firmware
+> SDK](https://github.com/golioth/golioth-firmware-sdk/tree/main/examples/zephyr)
+> that replaces it.
+
+### 1. Creating the Golioth client
+
+- The Golioth client now requires a `golioth_client_config` struct that
+  contains a credentials for authenticating with Golioth. This is used
+  for either certificate authentication or pre-shared key
+  authentication.
+
+    ```c
+    /* Config for Certificate authentication */
+    golioth_client_config_t client_config = {
+        .credentials = {
+            .auth_type = GOLIOTH_TLS_AUTH_TYPE_PKI,
+            .pki = {
+                .ca_cert = tls_ca_crt,
+                .ca_cert_len = sizeof(tls_ca_crt),
+                .public_cert = tls_client_crt,
+                .public_cert_len = tls_client_crt_len,
+                .private_key = tls_client_key,
+                .private_key_len = tls_client_key_len,
+            }}};
+
+    /* Config for PSK authentication */
+    golioth_client_config_t client_config = {
+        .credentials = {
+                .auth_type = GOLIOTH_TLS_AUTH_TYPE_PSK,
+                .psk = {
+                        .psk_id = client_psk_id,
+                        .psk_id_len = client_psk_id_len,
+                        .psk = client_psk,
+                        .psk_len = client_psk_len,
+                }}};
+    ```
+
+- Pass a pointer to the configuration when creating the Golioth client:
+
+    ```c
+    golioth_client_t client = golioth_client_create(&client_config);
+    ```
+
+- The callback that runs when Golioth connects no longer needs to be
+  registered, and now passes a `golioth_client_event` into the function:
```

Comment:
The callback does still need to be registered, but it is registered through a function call instead of setting a pointer in the struct:
`golioth_client_register_event_callback(client, on_client_event, NULL);`

Reply:
Ah, thanks for catching this. I've made an update to reference the new function call.